Scripts/RemediationScripts/Remediate-EnableWAFPolicyForFrontDoor.ps1

```diff
@@ -0,0 +1,956 @@
+                                                                                    $true
+                                                                                }
+                                                                                else
+                                                                                {
+                                                                                    $false
+                                                                                }
+                                                                            }
+                                                                        }}
+            }
+            catch
+            {
+                Write-Host "Error fetching Front Door FrontEndpoint:  ID - $($frontdoorFrontEndpointId). Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+                Write-Host "Skipping this Front Door FrontEndpoint..." -ForegroundColor $([Constants]::MessageType.Warning)
+            }
+        }
+    }
+    
+   
+     
+    if ([String]::IsNullOrWhiteSpace($FilePath))
+    {
+        $totalfrontDoors = ($frontDoors | Measure-Object).Count
+
+        if ($totalfrontDoors -eq 0)
+        {
+            Write-Host "No Front Doors found. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+            Write-Host $([Constants]::DoubleDashLine)
+            break
+        }
+
+        Write-Host "Found [$($totalfrontDoors)] Front Door(s)." -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host $([Constants]::SingleDashLine)
+    }
+    
+  
+   
+    $totalfrontDoorFrontendPoints = ($frontDoorFrontendPoints | Measure-Object).Count
+
+    if ($totalfrontDoorFrontendPoints -eq 0)
+    {
+        Write-Host "No Front Door Endpoint(s) found with WAF Policy not in enabled state. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        break
+    }
+    
+    Write-Host "Found [$($totalfrontDoorFrontendPoints)] Front Door Frontendpoint(s)" -ForegroundColor $([Constants]::MessageType.Update)
+    Write-Host $([Constants]::SingleDashLine)
+
+    # Includes Front Door Endpoint(s) where WAF Policy is not in Enabled State
+    $frontDoorEndpointsWithWAFPolicyNotInEnabledState = @()
+
+    # Includes Front Door Endpoint(s) that were skipped during remediation. There were errors remediating them.
+    $frontDoorEndpointsSkipped = @()
+    
+     
+    Write-Host "[Step 3 of 5] Fetching Frontendpoint(s) where WAF Policy is not in Enabled State"
+    Write-Host $([Constants]::SingleDashLine)
+    Write-Host "Separating Front Door Endpoint(s) for which WAF Policy is not in Enabled State..." -ForegroundColor $([Constants]::MessageType.Info)
+    Write-Host $([Constants]::SingleDashLine)
+
+    $frontDoorFrontendPoints | ForEach-Object {
+        $frontEndPoint = $_        
+            if(($_.IsWAFPolicyStateEnabled -eq $false) -and ($_.IsWAFConfigured -eq $true))
```

Comment:
What IF WAF itself is not configured?

Reply:
We have a seperate BRS for Configuring WAF Policy. [Remediate-ConfigureWAFPolicyForFrontDoor.ps1]. So we are just ignoring WAF Policies which are not configured. And Enabling state of Policies where WAF Policy is configured

Reply:
Can we at least show some warnings to the user in such case?